sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2341,6 +2341,118 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Returns a new DataFrame that contains the differences
```

Comment:
From reading the code I think the column order of the two datasets are unimportant (i.e. identical datasets with different column  orders are considered the  same).  I think this  is fine, but we should mention this somewhere in the doc as there might be some cases where you do care about column  order (e.g. if the ourput is to write some csv).

Reply:
The column order is described in the documentation of this method (rephrased it a bit):
```
   * The id columns are in order as given to the method. If no id columns are given then all
   * columns of this Dataset are id columns and appear in the same order. The remaining
   * non-id columns are in the order of this Dataset.
```

Reply:
yes- but that is just describing how the output is presented.  I think we should explicitly state that column order is not important when the diff is considering whether two rows are equal

Reply:
got it, will make that clear

Reply:
fixed